plugins/repository-azure/build.gradle

```diff
@@ -27,6 +27,7 @@ dependencies {
   compile 'com.microsoft.azure:azure-keyvault-core:1.0.0'
   compile 'com.google.guava:guava:20.0'
   compile 'org.apache.commons:commons-lang3:3.4'
+  testCompile project(':test:fixtures:azure-fixture')
```

Comment:
It would be better not to have multiple types of dependencies  on a project it makes things harder to change, we already depend on it as a fixture it would be nice to avoid depending on the jar as well. 
From what I see it's because the tests also use it to start up some fixtures, could we change that so they talk to one externally ? It would be even better if we could verify the retries as a unit tests without  talking to the actual service, e.x. by mocking the sdk ? I understand if that's more work that we don't want to take on right now, but it would be a nice to have.

Reply:
> It would be better not to have multiple types of dependencies on a project it makes things harder to change, we already depend on it as a fixture it would be nice to avoid depending on the jar as well.

Yes, I agree. I wanted to have the fixture plumbing and its logic located at the same place within the same project and this is the only way I know to do it. The `AzureHttpHandler` could be placed into the test framework (and the fixture depends on it) but I don't think this a better solution.

> From what I see it's because the tests also use it to start up some fixtures, could we change that so they talk to one externally ?

For `AzureBlobStoreRepositoryTests` having the server logic running in the same JVM is really convenient when it comes to understand or debug things. I guess we could change this to run on external servers but we'll lost the ability to debug easily.

> It would be even better if we could verify the retries as a unit tests without talking to the actual service, e.x. by mocking the sdk ?

This is what was done for a long time (and still done in some tests) but for `ESBlobStoreRepositoryIntegTestCase` and other `*RetriesTests` we want to test how the SDK behaves exactly in various scenarios of server side errors. The SDKs offer some way to mock the HTTP calls (either using Mockito or some kind of low-level execution mock) but it does not reflect the exact behavior of the SDK and issues like #46589 would have been hard to troubleshoot and fix.

Reply:
> I guess we could change this to run on external servers but we'll lost the ability to debug easily.

Thinking about this twice, we could maybe just always run the fixture within the container in debug mode and connect the IDE to the ephemeral port

Reply:
I'll leave this to your best judgment. I don't feel strongly about not having this dependency and you bring up good arguments for not doing it too.